### DISCUSSION TITLE
How can you retrieve the aliased field if your table has an alias with the traverser api?

### DISCUSSION THREAD

**Discussion by henry99923123** [jOOQ/jOOQ#15708] [Q&A] [answered]
Consider this query:

`context.select(Author.AUTHOR.LAST_NAME).from(Author.AUTHOR.as("a"));`

If you use the new Traverser API. How can you retrieve the LAST_NAME-Field that is aliased based on the aliased table? The input would be a normal field without the alias which has to be mapped to the aliased field.

The new API is great btw.

**Comment by lukaseder** [comment, marked as answer]
Thank you very much for your message.

I'm afraid I'm not sure what you mean by this. You didn't alias your field, so what do you mean by retrieving the aliased field? Do you mean your query is wrong, and how can you use the Traverser API to correct it? If that's what you mean, what would constitute a formal "correction"? While it looks "obvious" in this case, there seem to be many potential edge cases and ambiguities, as soon as you add joins/self-joins/derived tables, etc.
